### ISSUE TITLE
[Filebeat] [S3 input] should check object.size in handleSQSMessage

### ISSUE THREAD

**Issue by kwinstonix** [elastic/beats#19911]
- BUG

For confirmed bugs, please report:
- Version:  7.8.0
- Steps to Reproduce: 
  * create empty s3 object, so size is 0
  * run filebeat to process sqs message 

log output: 

```
ERROR	[s3]	s3/input.go:463	gzip.NewReader failed for 'test/xxxx.log.gz': EOF
```

if SQS record `s3.object.size`  is 0,  filebeat should stop processing this sqs message  and delete it .

**Comment by aleksmaus** [comment]
Did some testing.
Reproducable on 7.8 branch, not reproducable on master branch.
 
This looks to be fixed here
https://github.com/elastic/beats/commit/52765c423868dfa8d93e33997c69d41018a7ebbf#diff-ede3f69796ae7b221405174f187a241dR440

isStreamGzipped returns false for empty files and it skips gzip reader. 
Probably just needs to be backported to 7.x. Not sure what are the procedures.
Let me know if can help.

**Comment by exekias** [comment]
This was backported to 7.x (will be available in 7.9): https://github.com/elastic/beats/pull/18937. I guess we can close this?

**Comment by aleksmaus** [comment]
Awesome! Close and ship.
